### DISCUSSION TITLE
Cherry-pick selected commit to current branch like Merge selected branch to current one

### DISCUSSION THREAD

**Discussion by mestihudson** [jesseduffield/lazygit#3957] [Ideas]
Is there a way to cherry-pick directy from a selected commit hash?

Example, sometimes, I am working in a task, did some commits, but the other mates keep to merge into main branch.

Then a create a new branch, and listing the old working branch, I start to press ":" > git cherry-pick hash1, etc, etc...

It would be great if a could select that hash from old working branch, and, press, whatever,
"C", to cherry-pick that hash, how we can do it, now, about another branch, in relation to the current branch with "M".

**Comment by mark2185** [comment]
I'm not sure I quite understood your request, but have you tried [the current cherry-pick feature](https://github.com/jesseduffield/lazygit?tab=readme-ov-file#cherry-pick)?

**Comment by mestihudson** [comment]
I use this feature, @mark2185.
The idea is similar when you navigate across the branchs (diff the current) and you can press "M" to merge that selected to current.

In this scenario, you enter into a specific branch (diff the current), navigate across the commits, and you can press some shortcut key, to cherry-pick that commit into current branch.

Make sense?
